corehq/apps/commtrack/tests/test_locations.py

```diff
@@ -12,77 +17,74 @@ def _count_root_locations(domain):
     return SQLLocation.active_objects.root_nodes().filter(domain=domain).count()
 
 
-class LocationsTest(CommTrackTest):
-    user_definitions = [FIXED_USER]
+class LocationsTest(TestCase):
+    domain = 'westworld'
 
-    def setUp(self):
-        super(LocationsTest, self).setUp()
-        self.accessor = SupplyInterface(self.domain.name)
-        self.user = self.users[0]
+    @classmethod
+    def setUpClass(cls):
+        super(LocationsTest, cls).setUpClass()
+        delete_all_users()
+        cls.domain_obj = bootstrap_domain(cls.domain)
+        bootstrap_location_types(cls.domain)
 
-    @run_with_all_backends
-    def test_sync(self):
-        test_state = make_loc(
-            'teststate',
-            type='state',
-            parent=self.user.location
-        )
-        test_village = make_loc(
-            'testvillage',
-            type='village',
-            parent=test_state
-        )
+    @classmethod
+    def tearDownClass(cls):
+        super(LocationsTest, cls).tearDownClass()
+        cls.domain_obj.delete()  # domain delete cascades to everything else
+        delete_all_users()
 
-        try:
-            sql_village = SQLLocation.objects.get(
-                name='testvillage',
-                domain=self.domain.name,
-            )
+    def setUp(self):
+        super(LocationsTest, self).setUp()
+        self.user = CommCareUser.create(domain=self.domain, username='d.abernathy', password='123')
+        self.loc = make_loc('mariposa_saloon', domain=self.domain)
+        self.user.set_location(self.loc)
 
-            self.assertEqual(sql_village.name, test_village.name)
-            self.assertEqual(sql_village.domain, test_village.domain)
-        except SQLLocation.DoesNotExist:
-            self.fail("Synced SQL object does not exist")
+    def tearDown(self):
+        self.user.delete()
+        SQLLocation.objects.all().delete()
+        delete_all_docs_by_doc_type(Location.get_db(), ['Location'])
+        super(LocationsTest, self).tearDown()
```

Comment:
I didn't use `delete_all_locations` because I don't want to delete types (this is copy/pasted from there).  Types are defined in `setUpClass` and aren't modified by the tests.  They're cleaned up by `domain_obj.delete()`